packages/babel-helper-builder-react-jsx-experimental/src/index.js

```diff
@@ -303,12 +303,15 @@ You can set \`throwIfNamespace: false\` to bypass this warning.`,
       "JSXElement|JSXFragment"(path) {
         if (path.type === "JSXFragment") imports.add("Fragment");
         const openingPath = path.get("openingElement");
-        const validChildren = openingPath.parent.children.filter(
-          child =>
-            !t.isJSXEmptyExpression(child) &&
-            !(t.isJSXText(child) && child.value.trim() === ""),
-        );
-
+        const validChildren = openingPath.parent.children.filter(child => {
```

Comment:
Can we reuse [`t.react.buildChildren`](https://github.com/babel/babel/blob/master/packages/babel-types/src/builders/react/buildChildren.js)?

Reply:
`t.react.buildChildren` mutates the node. I can refactor it to return a value instead of mutating if that's okay with you.

Reply:
> t.react.buildChildren mutates the node.

Oh really? I thought it is side-effect free. I will double check later.

> I can refactor it to return a value instead of mutating if that's okay with you.

I am okay with that if it turns out we does not depend on its mutation behaviors. BTW `t.react.buildChildren` is undocumented so it is likely only used by our plugins.

Reply:
Ah actually maybe you're right. Let me update this diff